source/adminguide/systemvm.rst

```diff
@@ -848,3 +850,101 @@ same Debian 9 based templates.
 Non-Alphanumeric characters (metacharacters) are not allowed for this parameter
 except for the “-“ and the “.”. Any metacharacter supplied will immediately result
 in an immediate termination of the command and report back to the operator that an illegal character was passed
+
+Get Diagnostics Data
+~~~~~~~~~~~~~~~~~~~~
+
+For further troubleshooting, a set of files can be retrieved from any system VM
+by using the Get Diagnostics feature, either via the UI or an API call. The
+files are compressed and a URL is returned where the diagnostics data can be
+retrieved.
+
+#. As an administrator, log in to the CloudStack UI.
+
+#. Navigate to Infrastructure > System VMs or Virtual Routers.
+
+#. Click on the Get Diagnostics button. |get-diagnostics-icon.png|
+
+#. A form will pop up similar to this;
+
+      |diagnostics-data-form.png|
+
+#. Click OK.
+
+#. Wait for the URL to generate and click it to download the zipped up
+   diagnostics files.
+
+The following files are retrieved by default for each of the supported VMs and
```

Comment:
keep this seperately for each supported VM (Virtual Router, and Secondary Storage VM / Console Proxy VM), along with the respective global setting.